lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -431,10 +539,63 @@ sub add_batch_job {
     return $results;
 }
 
-
+$json_schemas{get_batch_job_result} = joi->object->strict->props(
+    batch_id => $Zonemaster::Backend::Validator::batch_id->required
+);
 sub get_batch_job_result {
     my ( $self, $batch_id ) = @_;
 
     return $self->{db}->get_batch_job_result($batch_id);
 }
+
+my $rpc_request = joi->object->props(
+    jsonrpc => joi->string->required,
+    id => joi->integer->min(0)->required,
+    method => joi->string->regex("[a-zA-Z0-9_-]*")->required
+);
+sub json_validate {
+    my ( $self, $json_schema) = @_;
+
+    my @error_rpc = $rpc_request->validate($json_schema);
+    return 1, {
+            jsonrpc => '2.0',
+            id => (defined $json_schema->{"id"} ? $json_schema->{'id'} + 0 : "1"),
+            error => {
+                code => '-32600',
+                message=> 'The JSON sent is not a valid Request object.',
+                data => "@error_rpc\n"
+            }
+        } if @error_rpc;
+
+    if (exists $json_schema->{"params"}) {
+
+        if ($json_schema->{"method"} eq "get_ns_ips" && ref \$json_schema->{"params"} eq "SCALAR") {
+            $json_schema->{"params"} = { domain => $json_schema->{"params"}};
+            warn "[DEPRECATE] - 'get_ns_ips' method using scalar is depreciated. Please update to {\"domain\"} \n";
+        } elsif ($json_schema->{"method"} eq "test_progress" && ref \$json_schema->{"params"} eq "SCALAR") {
+            $json_schema->{"params"} = { test_id => $json_schema->{"params"} };
+            warn "[DEPRECATE] - 'test_progress' method using scalar is depreciated. Please update to {\"test_id\"} \n";
+        } elsif ($json_schema->{"method"} eq "get_test_params" && ref \$json_schema->{"params"} eq "SCALAR") {
+            $json_schema->{"params"} = { test_id => $json_schema->{"params"} };
+            warn "[DEPRECATE] - 'get_test_params' method using scalar is depreciated. Please update to {\"test_id\"} \n";
+        } elsif ($json_schema->{"method"} eq "get_batch_job_result" && ref \$json_schema->{"params"} eq "SCALAR") {
+            $json_schema->{"params"} = { batch_id => $json_schema->{"params"} };
+            warn "[DEPRECATE] - 'get_batch_job_result' method using scalar is depreciated. Please update to {\"batch_id\"} \n";
+        } elsif ($json_schema->{"method"} eq "get_data_from_parent_zone" && ref \$json_schema->{"params"} eq "SCALAR") {
+            $json_schema->{"params"} = { domain => $json_schema->{"params"} };
+            warn "[DEPRECATE] - 'get_data_from_parent_zone' method using scalar is depreciated. Please update to {\"domain\"} \n";
+        }
+
+        my @error = $json_schemas{$json_schema->{"method"}}->validate($json_schema->{"params"});
+        return 1, {
+                jsonrpc => $json_schema->{"jsonrpc"},
+                id => $json_schema->{"id"},
+                result => {
```

Comment:
Why isn't this an error with code -32602?

Reply:
True ! Done in commit [d50e458](https://github.com/dotse/zonemaster-backend/pull/399/commits/d50e4581b0cf038f8d6ffce877983e4a441170b0)